test/testVegaWrapper.js

```diff
@@ -417,3 +418,351 @@ describe('vegaWrapper', function() {
+        wikiraw: ['wikiraw.nonsec.org', 'wikiraw.sec.org'],
+        wikidatasparql: ['wikidatasparql.nonsec.org', 'wikidatasparql.sec.org'],
+        geoshape: ['maps.nonsec.org', 'maps.sec.org']
+    };
+    var domainMap = {
+        'nonsec': 'nonsec.org',
+        'sec': 'sec.org'
+    };
+
+    var wrapper = new VegaWrapper2({
+            loader: {},
+            domains: domains,
+            domainMap: domainMap,
+            logger: function (msg) { throw new Error(msg); },
+            formatUrl: urllib.format,
+            languageCode: 'en'
+        });
+
+    describe('sanitize', function () {
+        var pass = function (url, expected, addCorsOrigin) {
+                var opt = {domain: 'domain.sec.org'};
+                const result = wrapper.objToUrl(url, opt);
+                assert.equal(result, expected, JSON.stringify(url));
+                assert.equal(opt.addCorsOrigin, addCorsOrigin, 'addCorsOrigin');
+            },
+            passWithCors = function (url, expected) {
+                pass(url, expected, true);
+            },
+            fail = function (url) {
+                expectError(function () {
+                    return wrapper.objToUrl(url, {domain: 'domain.sec.org'});
+                }, url, ['VegaWrapper2.objToUrl', 'VegaWrapper2._overrideHostAndProtocol']);
+            };
+
+        it('error type', function () {
+            fail({});
+            fail({ path: 'blah' });
+            fail({ type: 'blah', title: 'MyPage' });
+            fail({ type: 'nope', host: 'sec.org' });
+            fail({ type: 'nope', host: 'sec' });
+            fail({ type: 'https', host: 'sec.org' });
+            fail({ type: 'https', host: 'sec' });
+        });
+
+        it('wikiapi', function () {
+            // wikiapi allows sub-domains
+            fail({ type: 'wikiapi', wiki: 'sec.org', });
+            fail({ type: 'wikiapi', wiki: 'sec.org', params: 'blah'});
+            passWithCors({ type: 'wikiapi', params:{a: '1'} }, 'https://domain.sec.org/w/api.php?a=1&format=json&formatversion=2');
+            passWithCors({ type: 'wikiapi', wiki: 'sec.org', params:{a: 1} }, 'https://sec.org/w/api.php?a=1&format=json&formatversion=2');
+            passWithCors({ type: 'wikiapi', wiki: 'wikiapi.sec.org', params:{a: '1'} }, 'https://wikiapi.sec.org/w/api.php?a=1&format=json&formatversion=2');
+            passWithCors({ type: 'wikiapi', wiki: 'sec', params:{a: '1'} }, 'https://sec.org/w/api.php?a=1&format=json&formatversion=2');
+            passWithCors({ type: 'wikiapi', wiki: 'nonsec.org', params:{a: 1} }, 'http://nonsec.org/w/api.php?a=1&format=json&formatversion=2');
+            passWithCors({ type: 'wikiapi', wiki: 'wikiapi.nonsec.org', params:{a: '1'} }, 'http://wikiapi.nonsec.org/w/api.php?a=1&format=json&formatversion=2');
+            passWithCors({ type: 'wikiapi', wiki: 'nonsec', params:{a: '1'} }, 'http://nonsec.org/w/api.php?a=1&format=json&formatversion=2');
```

Comment:
need to also test for multiple params and multiple types, e.g. 
```
passWithCors({ type: 'wikiapi', params:{str: '1', num: 2, boolT: true, boolF: false },
'...?str=1&num=2&boolT=1');
```
note that `false` should simply skip the value.  Also we need a similar test to check proper URL encoding of symbols like &, %, and space.